### ISSUE TITLE
TS error when using imported Command type

### ISSUE THREAD

**Issue by marcopetreri** [tj/commander.js#1037]
As described in the example below I cannot use **Command** type in types definitions because of this error:
```
Exported variable 'getBuildCommand' has or is using name 'local.Command' from external module "***/node_modules/commander/typings/index" but cannot be named
```
Example
```typescript
import { Command } from 'commander';

export const getBuildCommand = (
   ...
) => (project: string, cmd: Command) => { ... };
```

**Comment by shadowspawn** [comment]
I was not able to reproduce this with typescript 3.6.2 and this program fragment. Someone else might recognise the error, or you could add some more context.

**Comment by fson** [comment]
I have made a minimal repoducible example for this issue: https://github.com/fson/commander-ts-repro.

When you compile the code using `npm run build`, it results in the following errror: 
> index.ts:3:1 - error TS4082: **Default export of the module has or is using private name 'local.Command'.**

**Comment by fson** [comment]
I also found a workaround: in some cases, the error went away after changing the exported function from an arrow function like this:
```ts
export default (program: Command) => { 
```
to a function expression:
```ts
export default function(program: Command) {
```

**Comment by shadowspawn** [comment]
Reproduced issue, and workaround, from @fson. (Thanks!)

Still trying to make sense of the local namespace pattern being used in the Commander definitions file.

**Comment by shadowspawn** [comment]
I've had a go at reworking the TypeScript declaration file and fixed this error, but the changes are tricky. If any TypeScript experts see this then a review would be very welcome.

**Comment by shadowspawn** [comment]
Fixed in Commander v4, which has been released.

**Comment by shadowspawn** [comment]
I was reproducing this again and a historical note. This issue can be reproduced using TypeScript 3.8.3, but is fixed from 3.9.2.
